### PR TITLE
feat: support authcode login in `kubectl oidc-login`

### PR DESCRIPTION
**Pull request by Unix4ever**
Omni now allows using `--grant-type=authcode-keyboard` option in `oidc-login`.
That allows to log in without forwarding `:8000` port from the remote machine. Instead kubelogin will present a link and ask for the one time code.

Clicking the link will open Omni web page with one time code.

Fixes: https://github.com/siderolabs/omni/issues/111

### CODE REVIEW

**Comment by Unix4ever** [comment]
The full flow looks like this

https://github.com/siderolabs/omni/assets/830532/079ae6f7-3902-4e17-9896-565ce4f32a85

File: client/pkg/omnictl/kubeconfig.go [hunk 1]
```diff
@@ -35,13 +35,22 @@ var kubeconfigCmdFlags struct {
...
 	serviceAccount       bool
 }
 
+var allGrantType = strings.Join([]string{
+	"auto",
+	"authcode",
+	"authcode-keyboard",
+	"password",
+	"device-code",
+}, "|")
```
Comment (utkuozdemir) [inline client/pkg/omnictl/kubeconfig.go:52]:
```suggestion
var allGrantTypes = strings.Join([]string{
	"auto",
	"authcode",
	"authcode-keyboard",
	"password",
	"device-code",
}, "|")
```

not all of these are working, right?

Comment (Unix4ever) [reply client/pkg/omnictl/kubeconfig.go:52]: Probably not, I haven't tried

Comment (Unix4ever) [reply client/pkg/omnictl/kubeconfig.go:52]: But the kubeconfig is valid with that, so I don't know

Comment (utkuozdemir) [reply client/pkg/omnictl/kubeconfig.go:52]: Should we validate against only the ones we know to be working? To avoid generating broken kubeconfigs?

Comment (Unix4ever) [reply client/pkg/omnictl/kubeconfig.go:52]: I guess that makes sense

**Review by utkuozdemir — APPROVED** [review]
🆒

File: client/pkg/omnictl/kubeconfig.go (hunk 1 above)
Comment (Unix4ever) [reply client/pkg/omnictl/kubeconfig.go]: updated. Left only `auto`, `authcode` and `authcode-keyboard`. Other 2 don't work

File: internal/backend/grpc/oidc.go [hunk 2]
```diff
@@ -49,11 +51,50 @@ func (s *oidcServer) Authenticate(ctx context.Context, req *oidc.AuthenticateReq
...
+func encodeToString(max int) (string, error) {
+	b := make([]byte, max)
+
+	n, err := io.ReadAtLeast(rand.Reader, b, max)
+	if n != max {
+		return "", err
+	}
+
+	for i := 0; i < len(b); i++ {
+		b[i] = table[int(b[i])%len(table)]
```
Comment (smira) [inline internal/backend/grpc/oidc.go:94]: this is not a exactly a good way, as %10 is biased towards some numbers more when the input is a byte (0..255). 

the easiest might be to have 16 characters (any power of 2 would work).

Comment (Unix4ever) [reply internal/backend/grpc/oidc.go:94]: added letters

**Pull request merged**